rosidl_typesupport_c/src/type_support_dispatch.hpp

```diff
@@ -63,25 +69,26 @@ get_typesupport_handle_function(
           fprintf(stderr, "Failed to find library '%s'\n", library_name);
           return nullptr;
         }
-        lib = new Poco::SharedLibrary(library_path);
+        rcutils_ret_t ret = rcutils_load_shared_library(lib, library_path.c_str());
+        if (ret != RCUTILS_RET_OK) {
+          fprintf(stderr, "Cannot open library %s", library_path.c_str());
```

Comment:
Again, throw, but this time probably `std::runtime_error` or a custom error based on that one.

Reply:
So looking at the rest of the function now, it's really strange that it uses fprintf in some places and throws exceptions in another.

I think we need to clean this up and either throw everywhere or catch all possible exceptions (like when doing `new`) and using fprintf/return nullptr in those cases too. My preference is to throw, since this is obviously a C++ function, as it is in a namespace and is a template function...

Reply:
We should check where it is being called from though, and make sure that place is setup to handle exceptions if/when they occur.

Reply:
Throwing `std exceptions` instead of using `fprintf`